File: 400. Nth Digit/solution.py
class Solution:
    def findNthDigit(self, n: int) -> int:
        for i in range(9):
            d = 9 * 10 ** i
            if n <= d * (i + 1):
                break
            n -= d * (i + 1)
        n -= 1
        return int(str(10**i + n / (i + 1))[n % (i + 1)])

File: 400. Nth Digit/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var findNthDigit = function(n) {
  let len = 1;
  let count = 9;
  let start = 1;

  while (n > len * count) {
    n -= len * count;
    len += 1;
    count *= 10;
    start *= 10;
  }

  start += (n - 1) / len;
  const s = start.toString();
  return s.charCodeAt((n - 1) % len) - "0".charCodeAt(0);
};
